include/zephyr/bluetooth/audio/bap.h

```diff
@@ -1019,6 +1033,28 @@ struct bt_bap_unicast_server_cb {
 	int (*release)(struct bt_bap_stream *stream, struct bt_bap_ascs_rsp *rsp);
 };
 
+/**
+ * @brief Register the Unicast Server.
+ *
+ * Register the Unicast Server.
+ * This will register ASCS in the GATT database.
+ *
+ * @param param  Registration parameters for ascs.
+ *
+ * @return 0 in case of success.
+ */
+int bt_bap_unicast_server_register(const struct bt_bap_unicast_server_register_param *param);
```

Comment:
The requirement that `bt_bap_unicast_server_register` shall be called after `bt_bap_unicast_server_register_cb` needs to be documented. 

If `bt_bap_unicast_server_unregister` is called after `bt_bap_unicast_server_register_cb`, does the user need to call `bt_bap_unicast_server_register_cb` again after a new `bt_bap_unicast_server_register`? 

 (this is why I want `bt_bap_unicast_server_register_cb` to only register callbacks and not all the other stuff it does)

Reply:
> The requirement that bt_bap_unicast_server_register shall be called after bt_bap_unicast_server_register_cb needs to be documented.

This has already been documented in the @brief of 'bt_bap_unicast_server_register_cb'

>If bt_bap_unicast_server_unregister is called after bt_bap_unicast_server_register_cb, does the user need to call bt_bap_unicast_server_register_cb again after a new bt_bap_unicast_server_register?

Yes, the user will have to do that. I will clarify the need for the user to strictly call unregister_cb before unregister too.

> (this is why I want bt_bap_unicast_server_register_cb to only register callbacks and not all the other stuff it does)

Yes, I really see your point here. Ill try to get work on this somewhere here after, so its updated before relase.

Reply:
>  I will clarify the need for the user to strictly call unregister_cb before unregister too.
Actually, I will change it so that 'bt_bap_unicast_server_unregister' will also unregister any callbacks.

Reply:
> Actually, I will change it so that 'bt_bap_unicast_server_unregister' will also unregister any callbacks.

`bt_bap_unicast_server_unregister` should only unregister callbacks if `bt_bap_unicast_server_register` registers them IMO.

Reply:
> This has already been documented in the @brief of 'bt_bap_unicast_server_register_cb'

`bt_bap_unicast_server_register_cb` has no reference to `bt_bap_unicast_server_register`, so not sure what you mean by this.

Reply:
> bt_bap_unicast_server_unregister should only unregister callbacks if bt_bap_unicast_server_register registers them IMO.

Fair enough, Ill make this change

> bt_bap_unicast_server_register_cb has no reference to bt_bap_unicast_server_register, so not sure what you mean by this.

Apparently I'm blind :D You are indeed correct

Reply:
I feel like it's still not documented that `bt_bap_unicast_server_register_cb` shall be called before `bt_bap_unicast_server_register`